Find string literals as a part of BinaryExpressions in jsx-no-literals

jaaberg opened the pull request:
Fixes #1511

jaaberg commented:
There exists tests for ``<Foo bar={`Test`} />`` and  ``<Foo>{\`Test\`}</Foo>``, but we could add one for ``<Foo bar={`Test ${baz}`} />`` to be explicit(?)

ljharb commented:
@jaaberg that'd be great; also `{`foo` + 'bar'}`, `{`foo` + `bar`}` and `{'foo' + `bar`}`?

jaaberg commented:
@ljharb Great, the tests actually uncovered another bug. The rule should find all strings/literals, so ``{`foo` + 'bar'}`` should actually give two errors, one for `foo` and another one for `bar`. 

Added tests in f27ebc2 and fixed in 28581df.
